annotation-processor/src/main/java/org/hibernate/validator/ap/checks/annotationparameters/AnnotationParametersScriptAssertCheck.java

```diff
@@ -23,25 +24,49 @@
  */
 public class AnnotationParametersScriptAssertCheck extends AnnotationParametersAbstractCheck {
 
-	public AnnotationParametersScriptAssertCheck(AnnotationApiHelper annotationApiHelper) {
+	private final Elements elementUtils;
+
+	public AnnotationParametersScriptAssertCheck(AnnotationApiHelper annotationApiHelper, Elements elementUtils) {
 		super( annotationApiHelper, TypeNames.HibernateValidatorTypes.SCRIPT_ASSERT );
+		this.elementUtils = elementUtils;
 	}
 
 	@Override
 	protected Set<ConstraintCheckIssue> doCheck(Element element, AnnotationMirror annotation) {
 		String lang = (String) annotationApiHelper.getAnnotationValue( annotation, "lang" ).getValue();
 		String script = (String) annotationApiHelper.getAnnotationValue( annotation, "script" ).getValue();
+		String reportOn = annotationApiHelper.getAnnotationValue( annotation, "reportOn" ) != null ?
+				(String) annotationApiHelper.getAnnotationValue( annotation, "reportOn" ).getValue() : "";
 		String alias = annotationApiHelper.getAnnotationValue( annotation, "alias" ) != null ?
 				(String) annotationApiHelper.getAnnotationValue( annotation, "alias" ).getValue() : "_this";
 
+		Set<ConstraintCheckIssue> issues = CollectionHelper.newHashSet();
 		if ( ( lang.trim().length() == 0 ) || ( script.trim().length() == 0 ) || ( alias.trim().length() == 0 ) ) {
-			return CollectionHelper.asSet(
+			issues.add(
 					ConstraintCheckIssue.error(
 							element, annotation, "INVALID_SCRIPT_ASSERT_ANNOTATION_PARAMETERS"
 					)
 			);
 		}
+		if ( !reportOn.isEmpty() ) {
+			boolean hasReportOnField = false;
+			for ( Element member : elementUtils.getAllMembers( (TypeElement) element ) ) {
+				if ( !member.getKind().isField() ) {
+					continue;
+				}
+				if ( member.getSimpleName().toString().equals( reportOn ) ) {
+					hasReportOnField = true;
+				}
+			}
+			if ( !hasReportOnField ) {
+				issues.add(
```

Comment:
It's a good consideration to check presence of the field, but actually it's not needed. `getPropertyNode()` in the node builder API doesn't mandate that the given property actually exists, it's intended to pass "fake fields", e.g. for map keys. So we shouldn't enforce this here.